src/ape/api/query.py

```diff
@@ -2,19 +2,46 @@
 
 import pandas as pd
 from ethpm_types.abi import EventABI, MethodABI
-from pydantic import BaseModel, NonNegativeInt, root_validator
+from pydantic import BaseModel, NonNegativeInt, root_validator, validator
 
-from ape._compat import Literal
 from ape.types import AddressType
 from ape.utils import BaseInterfaceModel, abstractmethod
 
+from .providers import BlockAPI
+from .transactions import TransactionAPI
+
 QueryType = Union["BlockQuery", "AccountQuery", "ContractEventQuery", "ContractMethodQuery"]
 
 
 class _BaseQuery(BaseModel):
-    type: str  # Used as discriminator
+
     columns: List[str]
 
+    @classmethod
+    @abstractmethod
+    def all_fields(cls) -> List[str]:
+        """
+        Validates fields that are called during a block query.
+
+        Returns:
+            List[str]: list of columns to be returned in pandas
+                dataframes during block query.
```

Comment:
```suggestion
            dataframes during block query.
```

^ This space (for next-line on returns doc) causes weird things to happen. Is fixed when not indented.